### PR TITLE
chore: reset and remove unsupported parameters in o1 model

### LINKED ISSUE #937: [Bug] OpenAI o1 models misalignment
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

o1 models don't support `None` as `tools` and `tool_choice` input and don't support system message

```
openai.BadRequestError: Error code: 400 - {'error': {'message': "Unsupported value: 'messages[0].role' does not support 'system' with this model.", 'type': 'invalid_request_error', 'param': 'messages[0].role', 'code': 'unsupported_value'}}

openai.BadRequestError: Error code: 400 - {'error': {'message': "Invalid type for 'tools': expected an unsupported value, but got null instead.", 'type': 'invalid_request_error', 'param': 'tools', 'code': 'invalid_type'}}
```

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mrsbeep**
## Description

Describe your changes in detail.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
`close #937 `

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/models/openai_model.py [hunk 1]
```diff
@@ -93,6 +93,20 @@ def run(
                 `ChatCompletion` in the non-stream mode, or
                 `Stream[ChatCompletionChunk]` in the stream mode.
         """
+        if self.model_type in [ModelType.O1_MINI, ModelType.O1_PREVIEW]:
```
Comment (Wendong-Fan) [inline camel/models/openai_model.py:96]:
```suggestion
        """
        # o1-preview and o1-mini have Beta limitations 
        # reference: https://platform.openai.com/docs/guides/reasoning
        if self.model_type in [ModelType.O1_MINI, ModelType.O1_PREVIEW]:
```

**Review by Wendong-Fan — COMMENTED** [review]
Thanks, @mrsbeep ! Since this issue is due to a limitation in the OpenAI beta version, I don't think we need to adjust the `ChatAgent` at the moment. We can temporarily handle it the same way you did with `OpenAIModel`. What do you think?

File: camel/models/openai_model.py (hunk 1 above)
Comment (mrsbeep) [reply camel/models/openai_model.py:96]: In fact, in all LLM models, including GPT, the system message is optional. The model can work without the system message. This is why I modified ChatAgent.

**Comment by mrsbeep** [comment]
In fact, in all LLM models, including GPT, the system message is optional. The model can work without the system message. This is why I modified ChatAgent.

File: camel/models/openai_model.py (hunk 1 above)
Comment (Wendong-Fan) [reply camel/models/openai_model.py]: I think it's important to define agent using system message, the change for `ChatAgent` need further discussion, how about we just keep the change focusing on model side and fix the bug first?

**Comment by Wendong-Fan** [comment]
> In fact, in all LLM models, including GPT, the system message is optional. The model can work without the system message. This is why I modified ChatAgent.

I think it's important to define agent using system message, the change for ChatAgent need further discussion, how about we just keep the change focusing on model side and fix the bug first?

**Comment by mrsbeep** [comment]
> > In fact, in all LLM models, including GPT, the system message is optional. The model can work without the system message. This is why I modified ChatAgent.
> 
> I think it's important to define agent using system message, the change for ChatAgent need further discussion, how about we just keep the change focusing on model side and fix the bug first?

Alright. I reset ChatAgent.

**Review by Wendong-Fan — APPROVED** [review]
Thanks @mrsbeep

**Pull request merged**